### ISSUE TITLE
Multi-class classification with fixed number of class for EBM

### ISSUE THREAD

**Issue by yap231995** [interpretml/interpret#398]
Hi, i have look at the source code and it seems like EBM do not consider fix num_classes for classification and just take the num_classes in the label but sometimes this is not the case.  
I required to have fix classes of 8 or 256 to compare with my neural network. Is there anyway to fix this or an option to change this?

**Comment by paulbkoch** [comment]
Hi @yap231995 -- EBMs currently support multiclass classification, but we do not natively support multitask or multilabel classification.  If you want to convert binary classifiers into a multi-label classifier, I believe the MultiOutputClassifier from scikit-learn over EBMs would work.

https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputClassifier.html#sklearn.multioutput.MultiOutputClassifier

**Comment by yap231995** [comment]
Hi Paul, thank you for the reply. Can we instead put a argument with `num_classes`, such that if i want to have a fix classes i could have instead of just getting the unique classes from the dataset. 

Changing from 

```
classes, y = np.unique(y, return_inverse=True)
n_classes = len(classes)
class_idx = {x: index for index, x in enumerate(classes)}
```

to 


``` 
if self.num_classes == None:
            classes, y = np.unique(y, return_inverse=True)
            n_classes = len(classes)
            class_idx = {x: index for index, x in enumerate(classes)}
else:
            _, y = np.unique(y, return_inverse=True)
            classes = np.array([x for x in range(self.num_classes)])
            n_classes = self.num_classes
            class_idx = {x: index for index, x in enumerate(classes)}
```

This will ensure I have a fix output size that I want.
Then we have argument in EBM class `num_classes` with `self.num_classes = num_classes`.

Can this be done?

**Comment by yap231995** [comment]
All I need is a fix number of class as sometimes my dataset will not give me the exactly same number of classes.
Is there a way for this to work?

**Comment by paulbkoch** [comment]
Hi @yap231995 -- I guess you could write a wrapper class to our class that would map your desired class labels to the labels in the trained model.  You would then do this conversion on calls to predict_proba, decision_function, etc..

Assuming there's a reasonable workaround, this strikes me a fairly niche scenario that doesn't warrant adding complexity to the __init__ function.  I don't believe any of the existing scikit-learn models handle this.  Perhaps a post-processing utility function that allows arbitrary re-ordering of the classes might be a better way to handle it.  We don't have support for anything like that though currently.  Open to receiving a PR on something like this though.
